### PR TITLE
[POAE7-2779] add string function split*, regexp_replace support

### PR DESCRIPTION
**Pull request by jikunshang**
### What changes were proposed in this pull request?

add string function split*, regexp_replace support,
leverage re2 library(instead of std::regex) to handle regular expression 

### Why are the changes needed?

support nextgen functionality

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

File: src/cider/exec/nextgen/function/CiderStringFunctions.cpp [hunk 1]
```diff
@@ -392,3 +395,218 @@ convert_string_to_time(const char* str_ptr, const int32_t str_len, const int32_t
...
+  size_t real_start_pos = (start_pos == npos) ? 0 : start_pos;
+  for (size_t i = real_start_pos; i >= str2_len; --i) {
+    if (!std::memcmp(str1 + i - str2_len, str2, str2_len)) {
+      return i - str2_len;
+    }
+  }
+  return npos;
+}
+
+extern "C" ALWAYS_INLINE int64_t cider_split(char* string_heap_ptr,
```
Comment (yma11) [inline src/cider/exec/nextgen/function/CiderStringFunctions.cpp:434]: For different branches in this function, are you referring any standards for handling different scenarios? If so, maybe you can give a link here. Same suggestion to other functions?

File: src/cider/exec/nextgen/function/CiderStringFunctions.cpp [hunk 2]
```diff
@@ -392,3 +395,218 @@ convert_string_to_time(const char* str_ptr, const int32_t str_len, const int32_t
...
+                      str_ptr,
+                      str_len,
+                      delimiter_ptr,
+                      delimiter_len,
+                      // shouldn't skip delimiter length on first search attempt
+                      delimiter_pos == 0 ? 0 : delimiter_pos + delimiter_len);
+    // do ++limit_counter in the loop to prevent bugs caused by shortcut execution
+    ++limit_counter;
+    // however, we still keep ++delimiter_idx in while condition check to ensure
+    // the property that delimiter_idx == 0 iff delimiter does not exist in input string
```
Comment (yma11) [inline src/cider/exec/nextgen/function/CiderStringFunctions.cpp:482]: typo `iff`.

**Review by yma11 — APPROVED** [review]
Overall LGTM.

File: src/cider/exec/nextgen/function/CiderStringFunctions.cpp (hunk 1 above)
Comment (jikunshang) [reply src/cider/exec/nextgen/function/CiderStringFunctions.cpp:434]: emmm, I referred HeavyDB implementation(which use std::string). Maybe I should list some examples here?

Comment (yma11) [reply src/cider/exec/nextgen/function/CiderStringFunctions.cpp:434]: Sure. Comments also can help.

**Review by yma11 — DISMISSED** [review]

**Review by kd97 — APPROVED** [review]

**Review by spevenhe — APPROVED** [review]

**Pull request merged**